contents/java/Java.md

```diff
@@ -14,6 +14,275 @@ Authors: John Doe, Jane Doe
+So you cannot access the age of your `Sheep` even though it is an `Animal`.
+
+This of course can be solved by reflections as follows:
+
+```java
+
+public class Sheep extends Animal {
+    private boolean hasWool;
+
+    public Sheep(String animalName) {
+        super(animalName);
+        hasWool = false;
+    }
+
+    @Override
+    public void onUpdate() {
+        super.onUpdate();
+
+        int age = getPrivateAge();
+
+        if (age > 20) {
+            setHasWool();
+        }
+    }
+
+    private void setHasWool() {
+        hasWool = true;
+    }
+
+    private int getPrivateAge() {
+        // Since class is a reserved keyword, we use clazz in variable names
+        Class<Animal> animalClazz = Animal.class;
+
+        try {
+            Field f = animalClazz.getDeclaredField("age");
+            f.setAccessible(true);
+            return (int) f.get(this);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        return 0;
+    }
+}
+```
+
+### Disadvantages of reflections
+
+While Java reflections is powerful, you should not immediately jump on the reflections ship. This is because there are
+some drawbacks whenever you use reflections in your project. The following are some points you should consider before
+using Java reflections:
+
+* Reflections converts a compile-time error to a potentially destructive run-time error.
+
+Compile time errors are easy to catch. Whenever you compile your code, the compiler cleverly spots any error you
+missed and points it out (along with line number and other useful information) to you before quitting. But by using
+reflections, you are bypassing these checks because there is no way to check such errors during compile time. Runtime
+errors are errors that you get during the operation of your software. For example you might have come across this problem
+where your program crashed and you get a `NullPointerException` error in your crash log. As you might have experienced
+already, runtime errors are more troublesome in that they are harder to catch and debug. They might even bring your whole
+software under the water with it by crashing the whole thing.
+
+* Reflections are harder to understand and harder to debug
+
+There is a reason why the topic of reflections is placed under the advanced section. Codes using reflections are 
+fundamentally harder to understand. As mentioned above, it is also harder to debug when the classes might not even be 
+there during compile time. This makes your code very hard to maintain.
+
+* Poor performance
+
+Since reflections resolve types dynamically, performance suffers. This is usually not an issue with small software
+but you might want to keep it in mind if you want to scale up.
+
+* Bad Security
+
+Remember at the beginning of the section, a part mentioned about accessing the private fields of a class using
```

Comment:
Maybe can change to reference your applications section since the example given there is more concrete?